cmd/marker/main.go

```diff
@@ -24,19 +24,28 @@ import (
 
 func main() {
 	nodeName := flag.String("node-name", "", "name of kubernetes node")
-	pollInterval := flag.Int("poll-interval", 60, "interval between updates in seconds, 60 by default")
+	updateInterval := flag.Int("update-interval", 60, "interval between updates in seconds, 60 by default")
+	pollInterval := flag.Int("poll-interval", 10, "interval between node info polls. every 10 cache updates by default")
 
 	flag.Parse()
 
 	if *nodeName == "" {
 		glog.Fatal("node-name must be set")
 	}
 
-	for {
-		err := marker.Update(*nodeName)
-		if err != nil {
-			glog.Errorf("Update failed: %v", err)
+	for i := 0; i <= *pollInterval; i++ {
+		useCache := true
+		if i == *pollInterval {
+			err := marker.Update(*nodeName, !useCache)
+			if err != nil {
+				glog.Errorf("Update failed: %v", err)
+			}
+		} else {
+			err := marker.Update(*nodeName, useCache)
+			if err != nil {
+				glog.Errorf("Update failed: %v", err)
+			}
+			time.Sleep(time.Duration(*updateInterval) * time.Second)
```

Comment:
Let's use the pollInmediate primitive here https://pkg.go.dev/k8s.io/apimachinery/pkg/util/wait#PollImmediate

Also the fact the useCache mechanism has to be implemented inside Update, usually a timestamp is stored about the cache refresh time, then elapsed is calculated between refresah timestamp and current timestamps and if elapsed is >= pollInterval then we refresh the cache before doing anything anything and we continue.